pkg/controller/seed-controller-manager/kubernetes/health.go

```diff
@@ -180,6 +196,42 @@ func (r *Reconciler) machineControllerHealthCheck(ctx context.Context, cluster *
 	}
 }
 
+func (r *Reconciler) operatingSystemManagerHealthCheck(ctx context.Context, cluster *kubermaticv1.Cluster, namespace string) (kubermaticv1.HealthStatus, error) {
+	// check for the health of operating-system-manager deployment.
+	key := types.NamespacedName{Namespace: namespace, Name: resources.OperatingSystemManagerDeploymentName}
+	status, err := resources.HealthyDeployment(ctx, r, key, 1)
+	if err != nil {
+		return kubermaticv1.HealthStatusDown, fmt.Errorf("failed to determine deployment's health %q: %w", resources.OperatingSystemManagerDeploymentName, err)
+	}
+
+	switch {
+	case status == kubermaticv1.HealthStatusUp:
+		return kubermaticv1.HealthStatusUp, nil
+	case status == kubermaticv1.HealthStatusProvisioning:
+		return kubermaticv1.HealthStatusProvisioning, nil
+	default:
+		return kubermaticv1.HealthStatusDown, nil
+	}
```

Comment:
Is this really necessary? This seems to be nothing more than `return status, nil`